### ISSUE TITLE
dependency problem ...

### ISSUE THREAD

**Issue by nauni77** [gematik/ref-eRp-FD-Server#11]
Hello,

I cloned the repository and build your software as described at your documentation. But now it receives a openssl 0.10.31 which causes the failure:

```
$ cargo build
############### removed content ... ##################
   Compiling actix-web v3.3.0 (https://github.com/Gematik-Entwicklung/actix-web.git?branch=ref-erx-fd-server-v3.3.0#39381387)
   Compiling vau v0.1.0 (/Users/wagnerol/development/noventi/gitRepos/10_NES/containerForNes/gematik_server.git/vau)
error[E0432]: unresolved import `openssl::hkdf`
  --> vau/src/decrypt.rs:24:5
   |
24 |     hkdf::{Hkdf, Mode},
   |     ^^^^ could not find `hkdf` in `openssl`

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> vau/src/decrypt.rs:77:13
   |
77 |         let aes_key = Hkdf::new(MessageDigest::sha256())?
   |             ^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: all local variables must have a statically known size
   = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> vau/src/decrypt.rs:77:23
   |
77 |           let aes_key = Hkdf::new(MessageDigest::sha256())?
   |  _______________________^
78 | |             .set_mode(Mode::ExtractAndExpand)?
79 | |             .set_info(Some(b"ecies-vau-transport"))?
80 | |             .set_secret(&shared_secret)?
81 | |             .derive(LEN_TAG)?;
   | |_____________________________^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: all local variables must have a statically known size
   = help: unsized locals are gated as an unstable feature

error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> vau/src/decrypt.rs:81:29
   |
81 |             .derive(LEN_TAG)?;
   |                             ^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `[u8]`
   = note: required by `std::prelude::v1::Err`

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0277, E0432.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `vau`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

If you set a fix dependency to openssl with `RUN cargo update -p openssl --precise v0.10.30` you can compile without failure.

Maybe you can fix your code that it workes with the latest version of openssl?

Best regards, Oli

**Comment by pandalec** [comment]
Hi Oli,

I am just curious, you have openssl-1.1.1 installed? And `which openssl` is pointing to the correct openssl installation? Because I am not running in any build problems.

Best regards,
parsi

**Comment by nauni77** [comment]
Hello, thanks for your response. I installed your application inside a container and build this container with:

```
RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y \
        build-essential \
        libssl1.1 \
        libssl-dev \
        pkg-config \
        git \
        curl \
        vim \
        && \
    rm -rf /var/lib/apt/lists/*
```
If I check for the version iside container it shows:

```
gematik@4ef70bb3d8dd:/application$ openssl version
OpenSSL 1.1.1f  31 Mar 2020
gematik@4ef70bb3d8dd:/application$
```

Something wrong with this?

Best regards,
Oli

**Comment by gematik1** [comment]
Hi Oli,
your dependency problem was fixed in current release 0.9.0. This was caused by a wrong dependency statement on rust-openssl fork.
 
Best regards,
Gematik Dev Team
